models/Asset/Document/ImageThumbnail.php

```diff
@@ -57,14 +57,25 @@ public function __construct($asset, $config = null, $page = 1, $deferred = true)
     }
 
     /**
-     * @param bool $deferredAllowed
+     * TODO: Pimcore 11: Change method signature to getPath($args = [])
+     *
+     * @param mixed $args,...
      *
      * @return string
      */
-    public function getPath($deferredAllowed = true)
+    public function getPath(...$args)
     {
+        // TODO: Pimcore 11: remove calling the covertArgsBcLayer() method
+        $args = $this->convertArgsBcLayer($args);
+
+        // set defaults
+        $deferredAllowed = $args['deferredAllowed'] ?? true;
+        $forceFrontend = $args['forceFrontend'] ?? false;
```

Comment:
Just one more input that came up by noticing that setting `forceFrontend` to `false` do not have any effects.

What if we rename it to a generic `$frontend` and use it to "override" a default behaviour when set, and leave the former behaviour as default fallback when is not set, eg.
```suggestion
        $frontend = $args['frontend'] ?? \Pimcore\Tool::isFrontend();
```
and removing line 77
`$frontend = \Pimcore\Tool::isFrontend() || $forceFrontend;`

in this way, one could even generate a path without frontend prefix in the frontend by setting `frontend` to `false`, not sure if anyone would need that but at least offers this option and it's a little more versatile and may be simpler to explain in the docs (compared to the reason behind "forcing" something)

Reply:
hi @kingjia90 lets do that. just have to push a new commit because the array key is "forceFrontend" not "frontend" and i've already commited your suggestion :P

Reply:
but that change from `forceFrontend` to `frontend` was on purpose to suggest a rename of this "parameter" key name as well, because it is IMHO not self-explainatory of what is forcing and/or would be hard to remember all the rules behind it.
While using `frontend`, knowing that the rule is that only in frontend this is by default true, then all would go along, one can set what one wants to get accordingly.


OT: noticed you are applying the same changes to the PR for 11.x branch, but JFYI that's not necessary overhead, we usually merge PR in 10.6, then we merge all 10.6 into 11.x, so in https://github.com/pimcore/pimcore/pull/14618 you would have just to rebase it and work on top of the changes former 10.6 but just have to wait the merges.

Reply:
got you! new commit is on the way.

thanks for the explanation on your branching strategy. that saves some time :)